specs/policies/network-address-translation.yaml

```diff
@@ -0,0 +1,483 @@
+        default: any
+    - name: nat-type
+      type: enum
+      profiles:
+        - xpath: [nat-type]
+      validators:
+        - type: values
+          spec:
+            values: [ipv4, nat64, nptv6]
+      spec:
+        values:
+          - { value: ipv4 }
+          - { value: nat64 }
+          - { value: nptv6 }
+    - name: to-interface
+      type: string
+      profiles:
+        - xpath: [to-interface]
+    - name: source-translation
+      type: object
+      profiles:
+        - xpath: [source-translation]
+      spec:
+        variants:
+          - name: dynamic-ip-and-port
+            type: string
+            profiles:
+              - xpath: [dynamic-ip-and-port]
+            validators:
+              - type: length
+                spec:
+                  max: 63
+          - name: interface-address
+            type: object
+            spec:
+              params:
+                - name: interface
+                  type: string
+                  profiles:
+                    - xpath: [interface-address]
+              variants:
+                - name: ip
+                  type: string
+                  profiles:
+                    - xpath: [ip]
+                - name: floating-ip
+                  type: string
+                  profiles:
+                    - xpath: [floating-ip]
+          - name: dynamic-ip
+            type: object
+            profiles:
+              - xpath: [dynamic-ip]
+            spec:
+              params:
+                - name: translated-address
+                  type: list
+                  profiles:
+                    - type: member
+                      xpath: [translated-address]
+                  spec:
+                    items:
+                      type: string
+                - name: fallback
+                  type: object
+                  profiles:
+                    - xpath: [fallback]
+                  spec:
+                    variants:
+                      - name: translated-address
+                        type: list
+                        profiles:
+                          - type: member
+                            xpath: [translated-address]
+                        spec:
+                          items:
+                            type: string
+                      - name: interface-address
+                        type: object
+                        profiles:
+                          - xpath: [interface-address]
+                        spec:
+                          params:
+                            - name: interface
+                              type: string
+                              profiles:
+                                - xpath: [interface]
+                              validators:
+                                - type: length
+                                  spec:
+                                    max: 31
+                            - name: ip
+                              type: string
```

Comment:
I think this one should be an object and have `spec.variants.{ip|floating_ip} of type string.

Reply:
We've discussed it during todays meeting: in XML API both ip and floating-ip are on the same level as interface, not wrapped in any other parent element. We've discussed adding this wrapping to terraform, but decided that wrapped version[1] would not be more readable.

[1] Wrapped versions would be one of:
```
destination_translation = {
  ip = {
    ip = "value"
    floating-ip = "or value"
  }
  interface = "ethernet1/1"
}
```
destination_translation = {
  ip = { # wrapper name is example, but couldn't figure out a better one during a meeting
    type = "ip" # or floating-ip
    value = "ip-value"
  }
  interface = "ethernet1/1"
}
```